### DISCUSSION TITLE
How to add an object with key:value pairs into an existing object in a schema

### DISCUSSION THREAD

**Discussion by jwalkerbg** [nlohmann/json#3482] [Q&A] [answered]
Here is my case. I have a validation schema, that have to receive key:value pairs from another json variable and put them, into its own existing properties object:

`json j1 = R"XYZ({
    "type":"object",
    "properties": {
        "tag": { "type": ["string","integer"]}
    }
})XYZ"_json;`

I have to add following content (example):
`json j2 = R"XYZ({
    "mode":{"type":"integer"},
    "strength":{"type":"integer"}
},
)XYZ"_json`

The final result should be:

`{
    "type":"object",
    "properties": {
        "tag": { "type": ["string","integer"]},
        "mode":{"type":"integer"},
        "strength":{"type":"integer"}
   }
}`

I tried to use
`j1["properties"] += j2;` but this rises exception `[json.exception.type_error.308] cannot use push_back() with object`.

Thank for the answers that would come,
Ivan

**Comment by nlohmann** [comment, marked as answer]
You may want to look at https://json.nlohmann.me/api/basic_json/update/.

**Comment by jwalkerbg** [comment]
Thank you very much, my test succeeded using  `update(j1,true)`, however I still had to use one intermediate object. Here is my test for reference:

`json j1 = R"XYZ({ "type":"object", "properties": { "tag": { "type": ["string","integer"]} } })XYZ"_json;`
`json j2 = R"XYZ({ "mode":{"type":"integer"}, "strength":{"type":"integer"} }, )XYZ"_json;`
`j3["properties"] = j2;`
`j1.update(j3,true);`

**Reply by falbrechtskirchinger** [reply]
Is `j1["properties"].update(j2)` an option?

**Reply by jwalkerbg** [reply]
Yes, it succeeded too, and it is the right solution, obviously. Thank you.
